libs/zmath/src/util.zig

```diff
@@ -0,0 +1,169 @@
+// ==============================================================================
+//
+// Collection of useful functions building on top of, and extending, core zmath.
+// https://github.com/michal-z/zig-gamedev/tree/main/libs/zmath
+//
+// ------------------------------------------------------------------------------
+// 1. Matrix functions
+// ------------------------------------------------------------------------------
+//
+// matTranslation(m: Mat) Vec
+// matForward(m: Mat) Vec
+// matUp(m: Mat) Vec
+// matRight(m: Mat) Vec
+//
+//
+// ------------------------------------------------------------------------------
+// 2. Angle functions
+// ------------------------------------------------------------------------------
+//
+// degToRad(angle: f32) f32
+// radToDeg(angle: f32) f32
+// angleMod(angle: f32) f32
+//
+// ==============================================================================
+
+const zm = @import("zmath.zig");
+const std = @import("std");
+const math = std.math;
+const expect = std.testing.expect;
+
+pub fn matTranslation(m: zm.Mat) zm.Vec {
+    return m[3];
```

Comment:
I think that we should set `w` component to 0 because user can take this vector and add to something and he may get unexpected result, for example:
```zig
const p0 = f32x4(1.0, 2.0, 3.0, 1.0); // point in space with w=1.0
const p1 = p0 + getTranslationVec(m); // now p1 has w==2.0 which is not expected
```